File: backend/app/schemas/job.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict

JobStatus = Literal["PENDING", "RUNNING", "WAITING_PROVIDER", "COMPLETED", "FAILED", "CANCELLED"]


class JobUpdate(BaseModel):
    status: JobStatus | None = None
    progress: int | None = None
    error_message: str | None = None


class JobRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: int
    clip_id: int | None
    job_type: str
    provider: str
    external_job_id: str | None
    status: str
    progress: int
    error_message: str | None
    retry_count: int
    created_at: datetime
    started_at: datetime | None
    completed_at: datetime | None

File: frontend/src/types/job.ts
export interface Job {
  id: number
  project_id: number
  clip_id: number | null
  job_type: string
  provider: string
  external_job_id: string | null
  status: string
  progress: number
  error_message: string | null
  retry_count: number
  created_at: string
  started_at: string | null
  completed_at: string | null
}
